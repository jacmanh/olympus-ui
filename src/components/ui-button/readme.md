# my-component



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type                                                                                         | Default     |
| -------- | --------- | ----------- | -------------------------------------------------------------------------------------------- | ----------- |
| `color`  | `color`   |             | `"blue" \| "green" \| "greenLight" \| "primary" \| "purple" \| "secondary" \| "transparent"` | `undefined` |
| `shape`  | `shape`   |             | `"circle"`                                                                                   | `undefined` |
| `size`   | `size`    |             | `"lg" \| "md" \| "sm" \| "xs"`                                                               | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
