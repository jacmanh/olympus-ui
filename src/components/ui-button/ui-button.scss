:host {
  display: inline-block;
  visibility: visible;
  color: #fff;
  border: transparent;
  border-radius: var(--border-radius-500);
  text-align: center;
  cursor: pointer;
}

:host([disabled]) {
  opacity: 0.2;
  cursor: not-allowed;
  background-color: var(--secondary-500) !important;
}

/* COLORS */
:host(.transparent) {
  color: var(--primary-500);
  background-color: transparent;
}
:host(.transparent:hover) {
  background-color: transparent;
}
:host(.transparent[disabled]) {
  background-color: transparent !important;
}

:host(.primary) {
  background-color: var(--primary-500);
}
:host(.primary:hover) {
  background-color: var(--primary-400);
}

:host(.secondary) {
  background-color: var(--secondary-500);
}
:host(.secondary:hover) {
  background-color: var(--primary-500);
}

:host(.blue) {
  background-color: var(--blue-500);
}
:host(.blue:hover) {
  background-color: var(--blue-400);
}

:host(.purple) {
  background-color: var(--purple-500);
}
:host(.purple:hover) {
  background-color: var(--purple-400);
}

:host(.green) {
  background-color: var(--green-500);
}
:host(.green:hover) {
  background-color: var(--green-400);
}

:host(.greenLight) {
  background-color: var(--greenLight-500);
}
:host(.greenLight:hover) {
  background-color: var(--greenLight-400);
}

/* SIZES */
:host(.xs) {
  font-size: var(--fontSize-xs);
  padding: var(--spacing-1);
}

:host(.sm) {
  font-size: var(--fontSize-sm);
  padding: var(--spacing-2);
}

:host(.md) {
  font-size: var(--fontSize-md);
  padding: var(--spacing-3);
}

:host(.lg) {
  font-size: var(--fontSize-lg);
  padding: var(--spacing-3);
}

/* SHAPES */
:host(.circle) {
  display: flex;
  width: 1em;
  height: 1em;
  overflow: hidden;
  place-items: center;
  justify-content: center;
  border-radius: 50%;
}
