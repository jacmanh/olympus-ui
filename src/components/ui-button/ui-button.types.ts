export enum ButtonColors {
  primary = 'primary',
  secondary = 'secondary',
  green = 'green',
  greenLight = 'greenLight',
  blue = 'blue',
  purple = 'purple',
  transparent = 'transparent',
}
export enum ButtonSizes {
  xs = 'xs',
  sm = 'sm',
  md = 'md',
  lg = 'lg',
}
export enum ButtonShapes {
  circle = 'circle',
}

export type ButtonColorProps = keyof typeof ButtonColors
export type ButtonSizeProps = keyof typeof ButtonSizes
export type ButtonShapesProps = keyof typeof ButtonShapes | null
