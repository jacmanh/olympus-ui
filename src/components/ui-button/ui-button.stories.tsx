import { ButtonColors, ButtonShapes, ButtonSizes } from './ui-button.types'

export default {
  title: 'Atom/ui-button',
  parameters: {},
}

interface ComponentProps {
  color: ButtonColors
  shape: ButtonShapes
  text?: string
}

const getComponent = ({ color = ButtonColors.primary, shape = null, text = null }: ComponentProps) => {
  return Object.keys(ButtonSizes)
    .map(
      key => `
        <div class="grid gap-3">
          <ui-button shape="${shape || ''}" color="${color}" size="${ButtonSizes[key]}">${text || `Button ${key}`}</ui-button>
          <ui-button shape="${shape || ''}" color="${color}" size="${ButtonSizes[key]}" disabled>${text || 'Disabled'}</ui-button>
        </div>
      `,
    )
    .join('')
}

const ListTemplate = (args: ComponentProps) => {
  return `<div class="flex items-start gap-3">
      ${getComponent(args)}
    </div>`
}

export const Primary = ListTemplate.bind({})
Primary.args = {
  name: 'Primary',
  color: 'primary',
}

export const Secondary = ListTemplate.bind({})
Secondary.args = {
  name: 'Secondary',
  color: 'secondary',
}

export const Green = ListTemplate.bind({})
Green.args = {
  name: 'Green',
  color: 'green',
}

export const GreenLight = ListTemplate.bind({})
GreenLight.args = {
  name: 'GreenLight',
  color: 'greenLight',
}

export const Blue = ListTemplate.bind({})
Blue.args = {
  name: 'Blue',
  color: 'blue',
}

export const Purple = ListTemplate.bind({})
Purple.args = {
  name: 'Purple',
  color: 'purple',
}

export const Transparent = ListTemplate.bind({})
Transparent.args = {
  name: 'Transparent',
  color: 'transparent',
}

export const Circle = ListTemplate.bind({})
Circle.args = {
  name: 'Circle',
  shape: 'circle',
  text: 'C',
}
