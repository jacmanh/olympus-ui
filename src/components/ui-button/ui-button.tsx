import { Component, h, Host, Prop } from '@stencil/core'
import { ButtonColorProps, ButtonShapesProps, ButtonSizeProps } from './ui-button.types'

@Component({
  tag: 'ui-button',
  styleUrl: 'ui-button.scss',
  shadow: true,
})
export class UiButton {
  @Prop() color: ButtonColorProps
  @Prop() size: ButtonSizeProps
  @Prop() shape: ButtonShapesProps

  getClassNames() {
    const color = this.color || 'primary'
    const size = this.size || 'md'
    const shape = this.shape || ''
    return [color, size, shape].join(' ')
  }

  render() {
    return (
      <Host class={this.getClassNames()}>
        <slot />
      </Host>
    )
  }
}
