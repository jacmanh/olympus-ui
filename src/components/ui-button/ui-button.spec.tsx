import { newSpecPage } from '@stencil/core/testing'
import { UiButton } from './ui-button'

describe('<ui-button />', () => {
  test('it should renders', async () => {
    const page = await newSpecPage({
      components: [UiButton],
      html: '<ui-button></ui-button>',
    })

    expect(page.root).toBeInTheDocument()
  })

  test('it should render with Primary styles (default)', async () => {
    const page = await newSpecPage({
      components: [UiButton],
      html: '<ui-button></ui-button>',
    })
    const button = page.root
    expect(button).toHaveClass('primary', 'md')
    expect(button).toMatchSnapshot()
  })

  test('it should render with Secondary styles', async () => {
    const page = await newSpecPage({
      components: [UiButton],
      html: '<ui-button color="secondary"></ui-button>',
    })
    const button = page.root
    expect(button).toHaveClass('secondary')
    expect(button).toMatchSnapshot()
  })

  test('it should render with "xs" styles', async () => {
    const page = await newSpecPage({
      components: [UiButton],
      html: '<ui-button size="xs"></ui-button>',
    })
    const button = page.root
    expect(button).toHaveClass('xs')
    expect(button).toMatchSnapshot()
  })

  test('it should render with "sm" styles', async () => {
    const page = await newSpecPage({
      components: [UiButton],
      html: '<ui-button size="sm"></ui-button>',
    })
    const button = page.root
    expect(button).toHaveClass('sm')
    expect(button).toMatchSnapshot()
  })
})
