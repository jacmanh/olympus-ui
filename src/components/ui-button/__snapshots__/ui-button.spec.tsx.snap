// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ui-button /> it should render with "sm" styles 1`] = `
<ui-button class="primary sm" size="sm">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</ui-button>
`;

exports[`<ui-button /> it should render with "xs" styles 1`] = `
<ui-button class="primary xs" size="xs">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</ui-button>
`;

exports[`<ui-button /> it should render with Primary styles (default) 1`] = `
<ui-button class="md primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</ui-button>
`;

exports[`<ui-button /> it should render with Secondary styles 1`] = `
<ui-button class="md secondary" color="secondary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
</ui-button>
`;
